L1Trigger/L1TTrackMatch/plugins/L1TrackJetEmulationProducer.cc

```diff
@@ -0,0 +1,671 @@
+            // otherwise, E0 is its own cluster
+            if (E2 < E1) {
+              L2cluster[nclust].pTtot += E1 + E2;
+              L2cluster[nclust].ntracks += ntrk1 + ntrk2;
+              L2cluster[nclust].nxtracks += nxtrk1 + nxtrk2;
+              L2cluster[nclust].phi = L1clusters[phibin + 1][used1].phi;
+              if (used1 >= 0)
+                L1clusters[phibin + 1][used1].used = true;
+              if (used2 >= 0)
+                L1clusters[phibin + 1][used2].used = true;
+              if (used3 >= 0)
+                L1clusters[phibin + 2][used3].used = true;
+              if (used4 >= 0)
+                L1clusters[phibin + 2][used4].used = true;
+            }
+            nclust++;
+            continue;
+          }  // end Not phiBins-2
+          else {
+            L2cluster[nclust].pTtot += E1;
+            L2cluster[nclust].ntracks += ntrk1;
+            L2cluster[nclust].nxtracks += nxtrk1;
+            L2cluster[nclust].phi = L1clusters[phibin + 1][used1].phi;
+            if (used1 >= 0)
+              L1clusters[phibin + 1][used1].used = true;
+            if (used2 >= 0)
+              L1clusters[phibin + 1][used2].used = true;
+            nclust++;
+            continue;
+          }
+        }       //End not last phibin(23)
+        else {  //if it is phibin 23
+          L1clusters[phibin][imax].used = true;
+          nclust++;
+        }
+      }  // while hipT not 0
+    }    // for each phibin
+
+    for (phibin = 0; phibin < phiBins_; ++phibin)
+      delete[] L1clusters[phibin];
+
+    // Now merge clusters, if necessary
+    for (int m = 0; m < nclust - 1; ++m) {
+      for (int n = m + 1; n < nclust; ++n) {
+        if (L2cluster[n].eta == L2cluster[m].eta &&
+            ((L2cluster[n].phi - L2cluster[m].phi < 3 * phiStep_ / 2 &&
+              L2cluster[m].phi - L2cluster[n].phi < 3 * phiStep_ / 2) ||
+             (L2cluster[n].phi - L2cluster[m].phi > 2 * l1t::Scales::makeGlbPhi(M_PI) - phiStep_ ||
+              L2cluster[m].phi - L2cluster[n].phi > 2 * l1t::Scales::makeGlbPhi(M_PI) - phiStep_))) {
+          if (L2cluster[n].pTtot > L2cluster[m].pTtot) {
+            L2cluster[m].phi = L2cluster[n].phi;
+          }
+            L2cluster[m].pTtot += L2cluster[n].pTtot;
+            L2cluster[m].ntracks += L2cluster[n].ntracks;
+            L2cluster[m].nxtracks += L2cluster[n].nxtracks;
+          for (int m1 = n; m1 < nclust - 1; ++m1) {
+            L2cluster[m1] = L2cluster[m1 + 1];
+          }
+          nclust--;
+          m = -1;
+          break;  //?????
+        }         // end if clusters neighbor in eta
+      }
+    }  // end for (m) loop
+    // sum up all pTs in this zbin to find ht
+
+    l1t::pt_t ht = 0;
+    for (int k = 0; k < nclust; ++k) {
+      if (L2cluster[k].pTtot > 50 && L2cluster[k].ntracks < lowpTJetMinTrackMultiplicity_)
+        continue;
+      if (L2cluster[k].pTtot > 100 && L2cluster[k].ntracks < highpTJetMinTrackMultiplicity_)
```

Comment:
Can the hardcoded numbers (50, 100) be defined globally or, better, taken as parameters?

Reply:
They're taken as parameters now.